lib/AST/SubstitutionMap.cpp

```diff
@@ -395,6 +398,40 @@ SubstitutionMap::combineSubstitutionMaps(const SubstitutionMap &firstSubMap,
     });
 }
 
+void SubstitutionMap::verify() const {
+#ifndef NDEBUG
```

Comment:
If the whole method is inside #ifndef, then all call-sites of this methods should be inside #ifndef as well. I find it a bit ugly.